diff --exclude=.git -x '*.orig' -x '*.rej' -Prc iTerm2/Interfaces/MainMenu.xib iTerm2-patched/Interfaces/MainMenu.xib
*** iTerm2/Interfaces/MainMenu.xib	2017-07-30 17:21:39.000000000 +1000
--- iTerm2-patched/Interfaces/MainMenu.xib	2017-07-30 17:17:23.000000000 +1000
***************
*** 1,5 ****
  <?xml version="1.0" encoding="UTF-8"?>
! <document type="com.apple.InterfaceBuilder3.Cocoa.XIB" version="3.0" toolsVersion="12121" systemVersion="16E195" targetRuntime="MacOSX.Cocoa" propertyAccessControl="none">
      <dependencies>
          <plugIn identifier="com.apple.InterfaceBuilder.CocoaPlugin" version="12121"/>
      </dependencies>
--- 1,5 ----
  <?xml version="1.0" encoding="UTF-8"?>
! <document type="com.apple.InterfaceBuilder3.Cocoa.XIB" version="3.0" toolsVersion="12121" systemVersion="16G29" targetRuntime="MacOSX.Cocoa" propertyAccessControl="none">
      <dependencies>
          <plugIn identifier="com.apple.InterfaceBuilder.CocoaPlugin" version="12121"/>
      </dependencies>
***************
*** 13,20 ****
          <customObject id="-3" userLabel="Application" customClass="NSObject"/>
          <menu title="MainMenu" systemMenu="main" id="29" userLabel="MainMenu">
              <items>
!                 <menuItem title="iTerm2" id="56">
!                     <menu key="submenu" title="iTerm2" systemMenu="apple" id="57">
                          <items>
                              <menuItem title="About iTerm2" identifier="About iTerm2" id="233">
                                  <connections>
--- 13,20 ----
          <customObject id="-3" userLabel="Application" customClass="NSObject"/>
          <menu title="MainMenu" systemMenu="main" id="29" userLabel="MainMenu">
              <items>
!                 <menuItem title="Terminal" id="56">
!                     <menu key="submenu" title="Terminal" systemMenu="apple" id="57">
                          <items>
                              <menuItem title="About iTerm2" identifier="About iTerm2" id="233">
                                  <connections>
***************
*** 1106,1126 ****
          </menu>
          <customObject id="201" userLabel="iTermApplicationDelegate" customClass="iTermApplicationDelegate">
              <connections>
-                 <outlet property="_buriedSessions" destination="PTu-aD-qbz" id="Fl9-Gn-jv2"/>
                  <outlet property="_showTipOfTheDay" destination="ft3-BP-igZ" id="WZt-qp-iuC"/>
-                 <outlet property="bookmarkMenu" destination="592" id="874"/>
                  <outlet property="closeTab" destination="635" id="795"/>
                  <outlet property="closeWindow" destination="598" id="796"/>
                  <outlet property="irPrev" destination="976" id="980"/>
-                 <outlet property="logStart" destination="229" id="793"/>
-                 <outlet property="logStop" destination="230" id="794"/>
                  <outlet property="maximizePane" destination="1233" id="1237"/>
                  <outlet property="selectTab" destination="847" id="863"/>
                  <outlet property="sendInputNormally" destination="1268" id="1276"/>
                  <outlet property="sendInputToAllPanes" destination="1266" id="1279"/>
                  <outlet property="sendInputToAllSessions" destination="1264" id="1278"/>
                  <outlet property="showFullScreenTabs" destination="1257" id="1260"/>
-                 <outlet property="toolbeltMenu" destination="1296" id="1299"/>
                  <outlet property="useTransparency" destination="1228" id="1230"/>
                  <outlet property="windowArrangementsAsTabs_" destination="z6V-EJ-Ntf" id="wse-Wc-WVw"/>
                  <outlet property="windowArrangements_" destination="1280" id="1285"/>
--- 1106,1121 ----
diff --exclude=.git -x '*.orig' -x '*.rej' -Prc iTerm2/iTerm2.xcodeproj/project.pbxproj iTerm2-patched/iTerm2.xcodeproj/project.pbxproj
*** iTerm2/iTerm2.xcodeproj/project.pbxproj	2017-07-30 17:21:39.000000000 +1000
--- iTerm2-patched/iTerm2.xcodeproj/project.pbxproj	2017-07-30 16:59:46.000000000 +1000
***************
*** 7199,7205 ****
  						DevelopmentTeam = H7V7XYVQ7D;
  					};
  					874206460564169600CFC3F1 = {
! 						DevelopmentTeam = H7V7XYVQ7D;
  						ProvisioningStyle = Automatic;
  					};
  					A66717851DCE36C3000CE608 = {
--- 7199,7205 ----
  						DevelopmentTeam = H7V7XYVQ7D;
  					};
  					874206460564169600CFC3F1 = {
! 						DevelopmentTeam = 4688J3MK2P;
  						ProvisioningStyle = Automatic;
  					};
  					A66717851DCE36C3000CE608 = {
***************
*** 8980,8986 ****
  				CODE_SIGN_IDENTITY = "Mac Developer";
  				COMBINE_HIDPI_IMAGES = YES;
  				COPY_PHASE_STRIP = YES;
! 				DEVELOPMENT_TEAM = H7V7XYVQ7D;
  				FRAMEWORK_SEARCH_PATHS = (
  					"$(inherited)",
  					"$(SRCROOT)",
--- 8980,8986 ----
  				CODE_SIGN_IDENTITY = "Mac Developer";
  				COMBINE_HIDPI_IMAGES = YES;
  				COPY_PHASE_STRIP = YES;
! 				DEVELOPMENT_TEAM = 4688J3MK2P;
  				FRAMEWORK_SEARCH_PATHS = (
  					"$(inherited)",
  					"$(SRCROOT)",
***************
*** 9121,9127 ****
  					"$(PROJECT_DIR)/ThirdParty",
  					"$(PROJECT_DIR)/ColorPicker",
  				);
! 				GCC_GENERATE_DEBUGGING_SYMBOLS = YES;
  				GCC_OPTIMIZATION_LEVEL = 3;
  				GCC_VERSION = "";
  				GCC_WARN_ABOUT_MISSING_PROTOTYPES = YES;
--- 9121,9128 ----
  					"$(PROJECT_DIR)/ThirdParty",
  					"$(PROJECT_DIR)/ColorPicker",
  				);
! 				GCC_FAST_MATH = YES;
! 				GCC_GENERATE_DEBUGGING_SYMBOLS = NO;
  				GCC_OPTIMIZATION_LEVEL = 3;
  				GCC_VERSION = "";
  				GCC_WARN_ABOUT_MISSING_PROTOTYPES = YES;
***************
*** 9210,9215 ****
--- 9211,9217 ----
  					"$(inherited)",
  					"\"$(SRCROOT)/ThirdParty/Growl.framework\"",
  				);
+ 				LLVM_LTO = YES;
  				MACOSX_DEPLOYMENT_TARGET = 10.10;
  				OTHER_CODE_SIGN_FLAGS = "--deep";
  				OTHER_LDFLAGS = (
***************
*** 10161,10167 ****
  				COMBINE_HIDPI_IMAGES = YES;
  				COPY_PHASE_STRIP = NO;
  				DEBUG_INFORMATION_FORMAT = dwarf;
! 				DEVELOPMENT_TEAM = H7V7XYVQ7D;
  				FRAMEWORK_SEARCH_PATHS = (
  					"$(inherited)",
  					"$(SRCROOT)",
--- 10163,10169 ----
  				COMBINE_HIDPI_IMAGES = YES;
  				COPY_PHASE_STRIP = NO;
  				DEBUG_INFORMATION_FORMAT = dwarf;
! 				DEVELOPMENT_TEAM = 4688J3MK2P;
  				FRAMEWORK_SEARCH_PATHS = (
  					"$(inherited)",
  					"$(SRCROOT)",
***************
*** 10170,10176 ****
  					"$(PROJECT_DIR)/ColorPicker",
  				);
  				GCC_DYNAMIC_NO_PIC = NO;
! 				GCC_GENERATE_DEBUGGING_SYMBOLS = YES;
  				GCC_OPTIMIZATION_LEVEL = 0;
  				GCC_VERSION = "";
  				GCC_WARN_ABOUT_MISSING_PROTOTYPES = YES;
--- 10172,10179 ----
  					"$(PROJECT_DIR)/ColorPicker",
  				);
  				GCC_DYNAMIC_NO_PIC = NO;
! 				GCC_FAST_MATH = YES;
! 				GCC_GENERATE_DEBUGGING_SYMBOLS = NO;
  				GCC_OPTIMIZATION_LEVEL = 0;
  				GCC_VERSION = "";
  				GCC_WARN_ABOUT_MISSING_PROTOTYPES = YES;
***************
*** 10196,10201 ****
--- 10199,10205 ----
  					"\"$(SRCROOT)/ThirdParty/Growl.framework\"",
  				);
  				LINK_WITH_STANDARD_LIBRARIES = YES;
+ 				LLVM_LTO = YES;
  				MACOSX_DEPLOYMENT_TARGET = 10.10;
  				"OTHER_CFLAGS[arch=*]" = "-DITERM_DEBUG=1";
  				OTHER_CODE_SIGN_FLAGS = "--deep";
***************
*** 10230,10236 ****
  				CODE_SIGN_IDENTITY = "Mac Developer";
  				COMBINE_HIDPI_IMAGES = YES;
  				COPY_PHASE_STRIP = YES;
! 				DEVELOPMENT_TEAM = H7V7XYVQ7D;
  				FRAMEWORK_SEARCH_PATHS = (
  					"$(inherited)",
  					"$(SRCROOT)",
--- 10234,10240 ----
  				CODE_SIGN_IDENTITY = "Mac Developer";
  				COMBINE_HIDPI_IMAGES = YES;
  				COPY_PHASE_STRIP = YES;
! 				DEVELOPMENT_TEAM = 4688J3MK2P;
  				FRAMEWORK_SEARCH_PATHS = (
  					"$(inherited)",
  					"$(SRCROOT)",
***************
*** 10238,10245 ****
  					"$(PROJECT_DIR)/ThirdParty",
  					"$(PROJECT_DIR)/ColorPicker",
  				);
! 				GCC_GENERATE_DEBUGGING_SYMBOLS = YES;
! 				GCC_OPTIMIZATION_LEVEL = 3;
  				GCC_VERSION = "";
  				GCC_WARN_ABOUT_MISSING_PROTOTYPES = YES;
  				GCC_WARN_FOUR_CHARACTER_CONSTANTS = NO;
--- 10242,10250 ----
  					"$(PROJECT_DIR)/ThirdParty",
  					"$(PROJECT_DIR)/ColorPicker",
  				);
! 				GCC_FAST_MATH = YES;
! 				GCC_GENERATE_DEBUGGING_SYMBOLS = NO;
! 				GCC_OPTIMIZATION_LEVEL = fast;
  				GCC_VERSION = "";
  				GCC_WARN_ABOUT_MISSING_PROTOTYPES = YES;
  				GCC_WARN_FOUR_CHARACTER_CONSTANTS = NO;
***************
*** 10263,10268 ****
--- 10268,10274 ----
  					"$(inherited)",
  					"\"$(SRCROOT)/ThirdParty/Growl.framework\"",
  				);
+ 				LLVM_LTO = YES;
  				MACOSX_DEPLOYMENT_TARGET = 10.10;
  				"OTHER_CFLAGS[arch=*]" = "";
  				OTHER_CODE_SIGN_FLAGS = "--deep";
diff --exclude=.git -x '*.orig' -x '*.rej' -Prc iTerm2/sources/PseudoTerminal.m iTerm2-patched/sources/PseudoTerminal.m
*** iTerm2/sources/PseudoTerminal.m	2017-07-30 17:21:39.000000000 +1000
--- iTerm2-patched/sources/PseudoTerminal.m	2017-07-30 16:59:46.000000000 +1000
***************
*** 635,641 ****
          [myWindow setFrame:initialFrame display:NO];
      }
  
!     [myWindow setHasShadow:(windowType == WINDOW_TYPE_NORMAL)];
  
      DLog(@"Create window %@", myWindow);
  
--- 635,641 ----
          [myWindow setFrame:initialFrame display:NO];
      }
  
!     [myWindow setHasShadow:YES]; // PATCH: (Modified) Native macOS window shadow
  
      DLog(@"Create window %@", myWindow);
  
diff --exclude=.git -x '*.orig' -x '*.rej' -Prc iTerm2/sources/iTermAdvancedSettingsModel.h iTerm2-patched/sources/iTermAdvancedSettingsModel.h
*** iTerm2/sources/iTermAdvancedSettingsModel.h	2017-07-30 17:21:39.000000000 +1000
--- iTerm2-patched/sources/iTermAdvancedSettingsModel.h	2017-07-30 16:59:46.000000000 +1000
***************
*** 23,28 ****
--- 23,31 ----
  + (NSString *)searchCommand;
  + (BOOL)dockIconTogglesWindow DEPRECATED_ATTRIBUTE;
  + (double)timeBetweenBlinks;
+ + (double)timeBeforeBlinks; // PATCH: (Added) Duration before cursor blinking starts
+ + (int)verticalCursorWidth; // PATCH: (Added) Vertical cursor width
+ + (BOOL)setCursorForegroundColor; // PATCH: (Added) Set cursor foreground color
  + (BOOL)neverWarnAboutMeta;
  + (BOOL)neverWarnAboutOverrides;
  + (BOOL)neverWarnAboutPossibleOverrides;
diff --exclude=.git -x '*.orig' -x '*.rej' -Prc iTerm2/sources/iTermAdvancedSettingsModel.m iTerm2-patched/sources/iTermAdvancedSettingsModel.m
*** iTerm2/sources/iTermAdvancedSettingsModel.m	2017-07-30 17:21:39.000000000 +1000
--- iTerm2-patched/sources/iTermAdvancedSettingsModel.m	2017-07-30 17:16:53.000000000 +1000
***************
*** 109,114 ****
--- 109,118 ----
  #pragma mark Terminal
  DEFINE_BOOL(traditionalVisualBell, NO, @"Terminal: Visual bell flashes the whole screen, not just a bell icon.");
  DEFINE_FLOAT(timeBetweenBlinks, 0.5, @"Terminal: Cursor blink speed (seconds).");
+ // PATCH: (Added) Duration before cursor blinking starts
+ DEFINE_FLOAT(timeBeforeBlinks, 0.5, @"Terminal: Duration before cursor starts blinking.");
+ // PATCH: (Added) Set cursor foreground color
+ DEFINE_BOOL(setCursorForegroundColor, YES, @"Terminal: Set cursor foreground color.");
  DEFINE_BOOL(doNotSetCtype, NO, @"Terminal: Never set the CTYPE environment variable.");
  // For these, 1 is more aggressive and 0 turns the feature off:
  DEFINE_FLOAT(smartCursorColorBgThreshold, 0.5, @"Terminal: Threshold for Smart Cursor Color for background color (0 to 1).\n0 means the cursor’s background color will always be the cell’s text color, while 1 means it will always be black or white.");
***************
*** 130,135 ****
--- 134,140 ----
  DEFINE_BOOL(zeroWidthSpaceAdvancesCursor, YES, @"Terminal: Zero-Width Space (U+200B) advances cursor?\nWhile a zero-width space should not advance the cursor per the Unicode spec, both Terminal.app and Konsole do this, and Weechat depends on it. You must restart iTerm2 after changing this setting.");
  DEFINE_BOOL(fullHeightCursor, NO, @"Terminal: Cursor occupies line spacing area.\nIf lines have more than 100% vertical spacing and this setting is enabled the bottom of the cursor will be aligned to the bottom of the spacing area.");
  DEFINE_FLOAT(underlineCursorOffset, 0, @"Terminal: Vertical offset for underline cursor.\nPositive values move it up, negative values move it down.");
+ DEFINE_INT(verticalCursorWidth, 2, @"Terminal: Width for vertical cursor.");
  DEFINE_BOOL(preventEscapeSequenceFromClearingHistory, NO, @"Terminal: Prevent CSI 3 J from clearing scrollback history?\nThis is also known as thethe terminfo E3 capability.");
  
  #pragma mark Hotkey
diff --exclude=.git -x '*.orig' -x '*.rej' -Prc iTerm2/sources/iTermCursor.m iTerm2-patched/sources/iTermCursor.m
*** iTerm2/sources/iTermCursor.m	2017-07-30 17:21:39.000000000 +1000
--- iTerm2-patched/sources/iTermCursor.m	2017-07-30 17:16:09.000000000 +1000
***************
*** 109,115 ****
               focused:(BOOL)focused
                 coord:(VT100GridCoord)coord
               outline:(BOOL)outline {
!     NSRect cursorRect = NSMakeRect(rect.origin.x, rect.origin.y, 1, rect.size.height);
      if (outline) {
          [self drawOutlineOfRect:cursorRect withColor:backgroundColor];
      } else {
--- 109,115 ----
               focused:(BOOL)focused
                 coord:(VT100GridCoord)coord
               outline:(BOOL)outline {
!     NSRect cursorRect = NSMakeRect(rect.origin.x, rect.origin.y, [iTermAdvancedSettingsModel verticalCursorWidth], rect.size.height); // PATCH: (Modified) Vertical cursor width
      if (outline) {
          [self drawOutlineOfRect:cursorRect withColor:backgroundColor];
      } else {
***************
*** 199,207 ****
                                       coord:coord];
          } else {
              // Non-smart
              [self.delegate cursorDrawCharacterAt:coord
                                       doubleWidth:doubleWidth
!                                    overrideColor:foregroundColor
                                           context:ctx
                                   backgroundColor:backgroundColor];
          }
--- 199,208 ----
                                       coord:coord];
          } else {
              // Non-smart
+             // PATCH: (Modified) Set cursor foreground color
              [self.delegate cursorDrawCharacterAt:coord
                                       doubleWidth:doubleWidth
!                                    overrideColor:[iTermAdvancedSettingsModel setCursorForegroundColor] == YES ? foregroundColor : nil
                                           context:ctx
                                   backgroundColor:backgroundColor];
          }
diff --exclude=.git -x '*.orig' -x '*.rej' -Prc iTerm2/sources/iTermRootTerminalView.m iTerm2-patched/sources/iTermRootTerminalView.m
*** iTerm2/sources/iTermRootTerminalView.m	2017-07-30 17:21:39.000000000 +1000
--- iTerm2-patched/sources/iTermRootTerminalView.m	2017-07-30 16:59:46.000000000 +1000
***************
*** 16,22 ****
  #import "iTermToolbeltView.h"
  #import "PTYTabView.h"
  
! const CGFloat kHorizontalTabBarHeight = 24;
  const CGFloat kDivisionViewHeight = 1;
  
  static const CGFloat kMinimumToolbeltSizeInPoints = 100;
--- 16,22 ----
  #import "iTermToolbeltView.h"
  #import "PTYTabView.h"
  
! const CGFloat kHorizontalTabBarHeight = 0; // PATCH: (Modified) No tab bar
  const CGFloat kDivisionViewHeight = 1;
  
  static const CGFloat kMinimumToolbeltSizeInPoints = 100;
diff --exclude=.git -x '*.orig' -x '*.rej' -Prc iTerm2/sources/iTermTextDrawingHelper.m iTerm2-patched/sources/iTermTextDrawingHelper.m
*** iTerm2/sources/iTermTextDrawingHelper.m	2017-07-30 17:21:39.000000000 +1000
--- iTerm2-patched/sources/iTermTextDrawingHelper.m	2017-07-30 16:59:46.000000000 +1000
***************
*** 2455,2464 ****
  }
  
  - (BOOL)shouldShowCursor {
      if (_cursorBlinking &&
          self.isInKeyWindow &&
          _textViewIsActiveSession &&
!         [NSDate timeIntervalSinceReferenceDate] - _lastTimeCursorMoved > 0.5) {
          // Allow the cursor to blink if it is configured, the window is key, this session is active
          // in the tab, and the cursor has not moved for half a second.
          return _blinkingItemsVisible;
--- 2455,2465 ----
  }
  
  - (BOOL)shouldShowCursor {
+     // PATCH: (Modified) Blink delay after moving the cursor 
      if (_cursorBlinking &&
          self.isInKeyWindow &&
          _textViewIsActiveSession &&
!         [NSDate timeIntervalSinceReferenceDate] - _lastTimeCursorMoved > [iTermAdvancedSettingsModel timeBeforeBlinks]) {
          // Allow the cursor to blink if it is configured, the window is key, this session is active
          // in the tab, and the cursor has not moved for half a second.
          return _blinkingItemsVisible;
